samples/valueentity-counter/src/main/java/com/example/domain/CounterImpl.java

```diff
@@ -25,7 +25,7 @@ public CounterImpl(@EntityId String entityId) { // <2>
 
     // tag::increase[]
     @Override
-    public Effect<Empty> increase(
+    public ValueEntityEffect<Empty> increase(
```

Comment:
yes, my plan is to remove the general `Effect` for exactly the reason you mention. If it's about the name `ValueEntityEffect` being too long we could make it a static inner class of `ValueEntity(Base)` but would be similar problem with accidental imports of wrong `Effect` as if it was a top level class `valueentity.Effect`.

Reply:
actually, it works without the import for a static inner class. So we can have `Effect` defined inside `ValueEntityBase` and then it can be used without any additional imports.

Reply:
that was neat: cbc8fac

Reply:
The imports will be need when writing tests though, but I think that's fine.